repos/system_upgrade/el7toel8/actors/postgresqlcheck/tests/test_postgresqlcheck.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+import pytest
+
+from leapp.libraries import stdlib
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.actor import library
+from leapp import reporting
+from leapp.reporting import Report, create_report
+from leapp.libraries.stdlib import api, run
+from leapp.models import RPM, InstalledSignedRPM
+
+RH_PACKAGER = 'Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>'
+
+
+class produce_mocked(object):
+    def __init__(self):
+        self.called = 0
+        self.model_instances = []
+
+    def __call__(self, *model_instances):
+        self.called += 1
+        self.model_instances.append(model_instances[0])
+
+
+class create_report_mocked(object):
+    def __init__(self):
+        self.called = 0
+        self.report_fields = None
+
+    def __call__(self, **report_fields):
+        self.called += 1
+        self.report_fields = report_fields
```

Comment:
The `produce_mocked` function is not ideal to test produced reports and it is duplicit code. Similar to `create_report_mocked`, which is not completely same, but pretty similar. Look at `create_report_mocked` in `libraries/testutils.py` instead. There are already some actors using it. And you can remove the `produce_mocked` completely as `produce` is part of the `create_report ` function.